code/game/objects/structures/beds_chairs/sofa.dm

```diff
@@ -4,6 +4,28 @@
 	icon = 'icons/obj/sofa.dmi'
 	buildstackamount = 1
 	item_chair = null
+	var/mutable_appearance/armrest
+
+/obj/structure/chair/sofa/Initialize()
+	armrest = mutable_appearance(icon, "[icon_state]_armrest", ABOVE_MOB_LAYER)
```

Comment:
We can improve this code by using a globally cached visual object, which will have its own layer.
Poke me if you need help with that.